### ISSUE TITLE
Build oneDAL from source

### ISSUE THREAD

**Issue by isuruf** [conda-forge/intel_repack-feedstock#87]
Since oneDAL is an open source project under the UXL foundation (a sub organization of the Linux foundation),
it would be good to build oneDAL from source instead of re-packaging the binary.

Thoughts on this @napetrov ?

**Comment by napetrov** [comment]
@isuruf  Yes, this was a longerterm plan to go with source build, starting with ARM, RISC-V builds.  First we need to cover valdiation for those-  now ARM validation is verry limited and need to be extended first through the upcoming year.
The first question would be on any potential issues of 2 feedstocks populating same package to different arches? 

On x86 build - this is more interesting. While x86 portion is more stable, oneDAL also covers SYCL based stuff which is less stable on one hand, have issues with public validation and we do have limitation on minimal requirements for SYCL there. 

And also we currently have intel driven cycles vs external to Intel release process - there are plans for establishing proper governance and decoupling this more from Intel.

**Comment by isuruf** [comment]
> On x86 build - this is more interesting. While x86 portion is more stable, oneDAL also covers SYCL based stuff which is less stable on one hand, have issues with public validation and we do have limitation on minimal requirements for SYCL there.

Is this about DPC++ version needed? I thought that we have all the dependencies in conda-forge to build oneDAL from source.

> And also we currently have intel driven cycles vs external to Intel release process - there are plans for establishing proper governance and decoupling this more from Intel.

Yep, given that it's governed by uxlfoundation, I'm hoping we don't have to rely on builds by Intel.

**Comment by napetrov** [comment]
> Is this about DPC++ version needed? I thought that we have all the dependencies in conda-forge to build oneDAL from source.

DPC++ requirements on glibc, but honestly we didn't look on current state for last 6 months.

> Yep, given that it's governed by uxlfoundation, I'm hoping we don't have to rely on builds by Intel.

Yes, but it's not a immediate thing - even now intel doing 99.8% contribution, which would be changing overtime and we would get more contribution, governance body. but this would take time. 


So i think we would start with non x86 builds to start new feedstock and then move more things there.   Might be one of the things that might work is decoupling of cpu and sycl implementaitons - this is somethign that we would discuss within oneDAL governance body. 
Also arm target feedstock would depend on non Intel contributors support.

**Comment by napetrov** [comment]
@isuruf  - we start discussion that potentially we could get non Intel source builds first.  Any considerations for scenario there 2 feedstocks would be pushing same packages but to different architectures?  i.e. repack would continue x86 deployments and separate onedal-feedstock would be deploying arm and risc?

**Comment by isuruf** [comment]
That's okay. We have to allow both feedstocks to upload the same name. We can build both x86_64 and aarch64 and upload them to a `conda-forge/label/onedal_dev`. We can run tests and then when we are sure, move them to `main`

**Comment by david-cortes-intel** [comment]
@isuruf Building oneDAL on windows currently requires `dos2unix`, which doesn't appear to be available for windows on conda-forge:
https://anaconda.org/conda-forge/dos2unix

Any chance to get a windows version or some type of replacement? It's used here for example:
https://github.com/uxlfoundation/oneDAL/blob/b33e7af41c85e79c10a893e54f436ed145af0baf/makefile#L1043

**Comment by jacobwilliams** [comment]
Also FYI: https://github.com/conda-forge/dos2unix-feedstock/issues/15

**Comment by david-cortes-intel** [comment]
PR adding an initial local-only conda recipe for x86 on linux was just merged in the oneDAL repository:
https://github.com/uxlfoundation/oneDAL/pull/3352

Perhaps it could be used as a starting point for an ARM recipe, which would need to remove DPC++ components and dependencies from it, plus add openblas.

**Comment by willend** [comment]
If `dos2unix` on `win_64` gives you any added mileage, FYI I have just updated the [feedstock](https://github.com/conda-forge/dos2unix-feedstock/) to include that platform.
